Core/Src/main.c

```diff
@@ -64,9 +64,16 @@ static void MX_TIM16_Init(void);
 
 /* Private user code ---------------------------------------------------------*/
 /* USER CODE BEGIN 0 */
-volatile uint32_t TxData[3];
+volatile uint64_t TxData[3];
 uint32_t bit = 0;
-uint32_t data = 0;
+
+uint64_t bit_data = 0;
+uint64_t bit_key = 0xD;
+uint64_t bit_appended;
+int shift = 60;
+int pos = 0;
+uint64_t bit_four = 0;
```

Comment:
generally directly modify the global variable is discouraged because it will be hard to track where the variable is modified in the program if the program gets big